api/client/sg.cc

```diff
@@ -422,20 +422,11 @@ static void sub_sg_rule_del_help(void) {
     global_parameter_help();
 }
 
-static int sub_sg_rule_del(int argc, char **argv,
-                            const GlobalOptions &options) {
-    if (argc >= 5 && string(argv[4]) == "help") {
-        sub_sg_rule_add_help();
-        return 0;
-    }
-
-    if (argc <= 5) {
-        sub_sg_rule_del_help();
-        return 1;
-    }
-
-    string sg = string(argv[4]);
-    string hash = string(argv[5]);
+static bool is_hash(string param) {
+    return param.length() == 16;
```

Comment:
a rule is an hexadecimal number, I would do something like: 
`return param.length() == 16 &&  isHex(param)`